################################################################################
SHORT DESCRIPTION:
################################################################################
Cast the NS_CERT_TYPE_APP macro

################################################################################
LONG DESCRIPTION:
################################################################################
NS_CERT_TYPE_APP is defined as a bitwise OR of values within 1-byte width.
With nss version 3.45, pesign compilation fails because of an implicit truncate
during the assignment. Since NS_CERT_TYPE_APP will always fit within a char
(1-byte), we can cast it on the RHS without worry.

################################################################################
CHANGELOG
################################################################################
Authors:
Chris Rogers <rogersc@ainfosec.com

################################################################################
REMOVAL
################################################################################
N/A

################################################################################
UPSTREAM PLAN
################################################################################
N/A

################################################################################
INTERNAL DEPENDENCIES
################################################################################
N/A

################################################################################
PATCHES
################################################################################

--- a/src/efikeygen.c
+++ b/src/efikeygen.c
@@ -206,7 +206,7 @@ static int
 add_cert_type(cms_context *cms, void *extHandle, int is_ca)
 {
 	SECItem bitStringValue;
-	unsigned char type = NS_CERT_TYPE_APP;
+	unsigned char type = (unsigned char)NS_CERT_TYPE_APP;
 
 	if (is_ca)
 		type |= NS_CERT_TYPE_SSL_CA |
